src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts

```diff
@@ -2647,30 +2739,8 @@ export abstract class AbstractTaskService extends Disposable implements ITaskSer
 			location: ProgressLocation.Window,
 			title: nls.localize('TaskService.fetchingBuildTasks', 'Fetching build tasks...')
 		};
-		let promise = this.getWorkspaceTasks().then(tasks => {
-			const buildTasks: (ConfiguringTask | Task)[] = [];
-			for (const taskSource of tasks) {
-				for (const task in taskSource[1].configurations?.byIdentifier) {
-					if (taskSource[1].configurations) {
-						const taskGroup: TaskGroup = taskSource[1].configurations.byIdentifier[task].configurationProperties.group as TaskGroup;
-
-						if (taskGroup && taskGroup._id === TaskGroup.Build._id && taskGroup.isDefault) {
-							buildTasks.push(taskSource[1].configurations.byIdentifier[task]);
-						}
-					}
-				}
-				if (taskSource[1].set) {
-					for (const task of taskSource[1].set?.tasks) {
-						const taskGroup: TaskGroup = task.configurationProperties.group as TaskGroup;
-						if (taskGroup && taskGroup._id === TaskGroup.Build._id && taskGroup.isDefault) {
-							buildTasks.push(task);
-						}
-					}
-				}
-				if (buildTasks.length > 0) {
-					break;
-				}
-			}
+		let promise = (async () => {
+			const buildTasks = await this._findWorkspaceTasksInGroup(TaskGroup.Build);
```

Comment:
This will only find default tasks. We still need to find the non-default build tasks for this command.